src/gl/state.js

```diff
@@ -0,0 +1,35 @@
+// @flow
+
+import type {Value} from './value';
+
+class State<T> {
+    value: Value<T>;
+    current: T;
+    dirty: boolean;
+
+    constructor(v: Value<T>) {
+        this.value = v;
+        this.current = v.constructor.default(v.context);
+    }
+
+    get(): T {
+        return this.current;
+    }
+
+    set(t: T) {
+        if (this.current !== t) {
```

Comment:
We have to be careful about using the `!==` operator here. It will always use object identity when `T` is an object or array type like `BlendFuncType` or `StencilFuncType`. That probably isn't what we want. We have a couple deep equal implementations that might work, or we could add an `equal(a: T, b: T): boolean` method to the `Value` interface.